File: lc_soln/0242_valid_anagram/solution.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) {
            return false;
        }
        var counterS = new Dictionary<char, int>();
        var counterT = new Dictionary<char, int>();
        foreach (char c in s) {
            counterS[c] = counterS.GetValueOrDefault(c) + 1;
        }
        foreach (char c in t) {
            counterT[c] = counterT.GetValueOrDefault(c) + 1;
        }
        foreach (KeyValuePair<char, int> kvp in counterS) {
            if (counterT.GetValueOrDefault(kvp.Key) != kvp.Value) {
                return false;
            }
        }
        foreach (KeyValuePair<char, int> kvp in counterT) {
            if (counterS.GetValueOrDefault(kvp.Key) != kvp.Value) {
                return false;
            }
        }
        return true;
    }
}

File: lc_soln/0242_valid_anagram/solution.py
from collections import Counter


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        counter_s = Counter(s)
        counter_t = Counter(t)
        for c, count in counter_s.items():
            if counter_t[c] != count:
                return False
        for c, count in counter_t.items():
            if counter_s[c] != count:
                return False
        return True


def main():
    print(Solution().isAnagram("racecar", "carrace"))


if __name__ == "__main__":
    main()
